docs/SingleNodeOpenShift.md

```diff
@@ -0,0 +1,29 @@
+# Single Node OpenShift
+
+Single Node OpenShift (SNO) clusters have a single control-plane node and are not considered HA, whereas OpenShift clusters with multiple control-plane nodes are considered HA as control-plane functions can be rescheduled on another node.
+
+## Differences
+
+There are several key differences in how the Machine Config Daemon handles non-HA/single node clusters:
+
+### Topology
+
+An annotation (`machineconfiguration.openshift.io/controlPlaneTopology`) is applied to all Node objects to describe the topology of the cluster. A non-HA/single node cluster is annotated with `SingleReplica`. In the absence of this annotation, the Machine Config Daemon defaults to `HighlyAvailable`. This primarily affects the node drain behavior as discussed below.
```

Comment:
MCO reads `controlPlaneTopology` value set in [infrastructure](https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_infrastructure.crd.yaml#L182) cluster object and sets that value into `controllerconfig`. Later on, Node controller reads `controlPlaneTopology` value from `controllerconfig` and adds an annotation (`machineconfiguration.openshift.io/controlPlaneTopology`) to all Node objects to describe the `controlPlaneTopology` of the cluster. A non-HA/single node cluster is annotated with `SingleReplica`. In the absence of this annotation, the Machine Config Daemon defaults to `HighlyAvailable`. This primarily affects the node drain behavior as discussed below.
```suggestion
An annotation (`machineconfiguration.openshift.io/controlPlaneTopology`) is applied to all Node objects to describe the topology of the cluster. A non-HA/single node cluster is annotated with `SingleReplica`. In the absence of this annotation, the Machine Config Daemon defaults to `HighlyAvailable`. This primarily affects the node drain behavior as discussed below.
```

Reply:
Done!